libs/sdk-ui-kit/src/Dialog/StylingEditorDialog/StylingEditorDialog.tsx

```diff
@@ -64,32 +63,51 @@ const StylingEditorDialogCore = <T extends StylingPickerItemContent>(props: ISty
         disableSubmit,
         showProgressIndicator,
     } = props;
+    const intl = useIntl();
     const providedExamples = !!examples && examples.length !== 0 && !!exampleToColorPreview;
-    const [nameField, setNameField] = useState(stylingItem?.name ?? "");
+    const initialNameField = stylingItem?.name ?? "";
+    const initialDefinitionField = stylingItem?.content ? JSON.stringify(stylingItem?.content, null, 4) : "";
+    const [nameField, setNameField] = useState(initialNameField);
+    const [definitionField, setDefinitionField] = useState(initialDefinitionField);
 
-    const [definitionField, setDefinitionField] = useState(
-        stylingItem?.content ? JSON.stringify(stylingItem.content, null, 4) : "",
-    );
-    const [invalidDefinition, setInvalidDefinition] = useState(false);
-    const isSubmitDisabled = useMemo(
-        () => nameField === "" || definitionField === "" || invalidDefinition || disableSubmit,
-        [nameField, definitionField, invalidDefinition, disableSubmit],
-    );
-    const intl = useIntl();
-    const validateDefinition = (value: string): boolean => {
+    const fieldsChanged = useMemo(() => {
+        try {
+            const parsedDefinition = JSON.parse(definitionField);
+            const formattedDefinition = JSON.stringify(parsedDefinition, null, 4);
+            return nameField !== initialNameField || formattedDefinition !== initialDefinitionField;
+        } catch (e) {
+            // initial state of the fields is presumed to be valid,
+            // so if JSON throws error, definition was changed
+            return true;
+        }
+    }, [nameField, definitionField]);
+
+    const validDefinition = useMemo(() => {
         try {
-            JSON.parse(value);
+            JSON.parse(definitionField);
```

Comment:
this fails also on empty input "", so below check for || definitionField === "" is not needed